menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/batch_norm.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef MENOH_IMPL_MKLDNN_WITH_GENERIC_FALLBACK_BACKEND_BACKEND_MKLDNN_OPERATOR_BATCH_NORM_HPP
+#define MENOH_IMPL_MKLDNN_WITH_GENERIC_FALLBACK_BACKEND_BACKEND_MKLDNN_OPERATOR_BATCH_NORM_HPP
+
+#include <menoh/mkldnn_with_generic_fallback/backend/mkldnn/formatted_array.hpp>
+#include <menoh/mkldnn_with_generic_fallback/backend/mkldnn/memory_cache.hpp>
+#include <menoh/mkldnn_with_generic_fallback/backend/mkldnn/operator/output_management.hpp>
+#include <menoh/mkldnn_with_generic_fallback/backend/mkldnn/procedure_factory.hpp>
+
+#include <mkldnn.hpp>
+
+namespace menoh_impl {
+    namespace mkldnn_with_generic_fallback_backend {
+        namespace mkldnn_backend {
+
+            inline procedure_factory_return_type make_batch_norm(
+              MENOH_MKLDNN_CONTEXT_PROCEDURE_FACTORY_PARAMETER_LIST) {
+
+                std::vector<mkldnn::primitive> primitives;
+                std::vector<std::pair<std::string, memory_cache>>
+                  temp_memory_cache_list;
+
+                auto epsilon = attribute_float(node, "epsilon");
+                auto spatial = attribute_int(node, "spatial");
+                if(!spatial) {
+                    throw std::runtime_error(
+                      "Non spacial BatchNorm is not supported");
```

Comment:
It might be better to use official operator name "BatchNormalization".

Reply:
I see. I will fix.